src/client/renderer/r_array.c

```diff
@@ -180,6 +182,12 @@ void R_UploadToSubBuffer(r_buffer_t *buffer, const size_t start, const size_t si
 		return;
 	}
 
+	// Don't allow null ptrs since bufferSubData does not allow it.
+	if (!data) {
+		Com_Warn("Attempted to upload null to GPU\n");
```

Comment:
The problem is that as a programmer I didn't even notice it because of how specific the issue came up in. Since it's a potentially fatal error it should be something that catches their eyes immediately.